main.py

```diff
@@ -97,6 +98,41 @@ def get_plugin_link_mapping(config, access_token):
     data_sets = list_response.json()['BrightspaceDataSets']
     return { d['PluginId']: d['DownloadLink'] for d in data_sets }
 
+def get_number_of_columns(db_conn_params, table):
+    with psycopg2.connect(**db_conn_params) as conn:
+        with conn.cursor() as cur:
+            cur.execute(
+                sql.SQL('''
+                    SELECT *
+                    FROM {table}
+                    LIMIT 0;
+                ''')
+                .format(
+                    table=sql.Identifier(table)
+                )
+            )
+
+            return len(cur.description)
+
+def process_csv_stream(csv_input_stream, num_columns_in_table):
+    '''
+    Ignore excessive columns in the CSV due to additive changes / BDS minor
+    changes by ignoring any columns in the CSV past the number of columns in the
+    table
+    '''
+    csv_rows = []
+    csv_reader = csv.reader(csv_input_stream, quoting=csv.QUOTE_MINIMAL)
+    for line in csv_reader:
```

Comment:
Do we ever express caveats with customer examples: for example this code may not be performant for very large CSV files, etc.